bottlerocket/tests/workload/nvidia-smoke/README.md

```diff
@@ -0,0 +1,68 @@
+# NVIDIA GPU Smoke tests
+
+The image created by the Dockerfile in this folder compiles a few [CUDA samples](https://github.com/NVIDIA/cuda-samples/) for both `x86_64` and `aarch64`.
+This container image can be used as a TestSys workload test to validate the Bottlerocket `*-nvidia` variants running on NVIDIA hosts.
+
+## Build requirements
+
+Due to missing packages needed for cross-compiling, and unexpected handling in the sample Makefile logic when compiling for one platform for another, the build for this image requires two Buildx hosts - one for each architecture.
+
+In order to build the image for both architectures, you need to set up the `buildx` docker plugin.
+[This guide describes](https://github.com/docker/buildx#installing) the installation process.
+
+By default, the install of `docker buildx` uses the `docker` driver type.
+This driver is not able to build for all platform targets we need.
+You will need to run the following to create a new builder using the `docker-container` driver:
+
+```bash
+docker buildx create --use --bootstrap
```

Comment:
This environment set up and subsequent 15 steps make me sad. :cry: 


Do I understand correctly that you are running another machine on the target architecture? Wow.
Not your fault though, cross compilation of containers is what it is...

Thank you for documenting this well.

Reply:
Yeah, that's how I originally had it - with one host. It made cross-compiling incredibly slow.

And it was working for me running from an arm64 host fine, but when I tried running from an amd64 host I would get weird compile/linking errors with the NVIDIA sample binaries. At first I chalked this up to something odd with building for ARM on AMD, but then after cleaning up and resetting things I started to get the same errors on my arm64 host that had been previously working.

So I think in general it is fine to do both from one host, but in this particular case there is something about how the Makefiles do various checks and set variables that control compilation that somehow gets messed up when using the normal QEMU based route.

Unfortunately the only way I could find to reliably get things to build right was the currently documented approach of using two different machines. I'm not happy about it, but it's the best I could come up with.

It does have the nice benefit that compilation is much faster than the previous emulated approach - by probably 5x. So there is some setup pain to do it this way, but it kind of pays off in other ways.